src/odbc.cpp

```diff
@@ -15,797 +15,842 @@
-          );
-
-          Nan::ThrowError(objError);
-          return scope.Escape(Nan::Undefined());
-          break;
-        }
-      } while (true);
-      
-      return scope.Escape(str);
+    data->storedRows.push_back(row);
   }
 }
 
-/*
- * GetRecordTuple
- */
+// All of data has been loaded into data->storedRows. Have to take the data
+// stored in there and convert it it into JavaScript to be given to the
+// Node.js runtime.
+Napi::Array ODBC::ProcessDataForNapi(Napi::Env env, QueryData *data) {
 
-Local<Value> ODBC::GetRecordTuple ( SQLHSTMT hStmt, Column* columns, 
-                                         short* colCount, uint16_t* buffer,
-                                         int bufferLength) {
-  Nan::EscapableHandleScope scope;
-  
-  Local<Object> tuple = Nan::New<Object>();
-        
-  for(int i = 0; i < *colCount; i++) {
-#ifdef UNICODE
-    tuple->Set( Nan::New((uint16_t *) columns[i].name).ToLocalChecked(),
-                GetColumnValue( hStmt, columns[i], buffer, bufferLength));
-#else
-    tuple->Set( Nan::New((const char *) columns[i].name).ToLocalChecked(),
-                GetColumnValue( hStmt, columns[i], buffer, bufferLength));
-#endif
+  std::vector<ColumnData*> *storedRows = &data->storedRows;
+  Column *columns = data->columns;
+  int columnCount = data->columnCount;
+
+  Napi::Array rows = Napi::Array::New(env);
+  if (storedRows->size() > 0) {
+    rows.Set(Napi::String::New(env, COUNT), Napi::Number::New(env, storedRows->size()));
+  } else {
+    rows.Set(Napi::String::New(env, COUNT), Napi::Number::New(env, data->rowCount));
   }
-  
-  return scope.Escape(tuple);
+
+  // attach the column information
+  Napi::Array napiColumns = Napi::Array::New(env);
+
+  for (SQLSMALLINT h = 0; h < columnCount; h++) {
+    Napi::Object column = Napi::Object::New(env);
+    column.Set(Napi::String::New(env, NAME), Napi::String::New(env, (const char*)columns[h].name));
+    column.Set(Napi::String::New(env, DATA_TYPE), Napi::Number::New(env, columns[h].type));
+
+    napiColumns.Set(h, column);
+  }
+
+  rows.Set(Napi::String::New(env, COLUMNS), napiColumns);
+
+  for (size_t i = 0; i < storedRows->size(); i++) {
+
+    // Arrays are a subclass of Objects
+    Napi::Object row = Napi::Object::New(env);
+
+    ColumnData *storedRow = (*storedRows)[i];
+
+    // Iterate over each column, putting the data in the row object
+    for (int j = 0; j < columnCount; j++) {
+
+      Napi::Value value;
+
+      // check for null data
+      if (storedRow[j].size == SQL_NULL_DATA) {
+
+        value = env.Null();
+
+      } else {
+
+        switch(columns[j].type) {
+          case SQL_REAL:
+          case SQL_NUMERIC :
+            value = Napi::Number::New(env, atof((const char*)storedRow[j].data));
+            break;
+          // Napi::Number
+          case SQL_DECIMAL :
+          case SQL_FLOAT :
+          case SQL_DOUBLE :
+            value = Napi::Number::New(env, *(double*)storedRow[j].data);
```

Comment:
Hi,

Shouldn't this line convert the data to `double` instead of just casting it ?
I tried to use this (the last version of `NAPI_sandbox`) and I was just having random numbers (e.g `3.52641852e28`).

I replaced it with this conversion and it works for me!
```suggestion
            value = Napi::Number::New(env, strtod((const char*)storedRow[j].data, NULL));
```

I can make a PR if needed :wink:! I think that the same will be needed for int...

Anyway, nice work!